### PR TITLE
Update Custom Hostnames API + add to CLI

### PR DESCRIPTION
**Pull request by elithrar**
- [x] Update Custom Hostnames API to support pagination, List vs. Filter.
- [x] Update tests
- [ ] Add full support for Custom Hostnames ("SSL for SaaS") to flarectl (CLI)

### CODE REVIEW

**Comment by elithrar** [comment]
(Work in progress)

File: custom_hostname.go [hunk 1]
```diff
@@ -13,22 +13,25 @@ type CustomHostnameSSL struct {
...
 
 // CustomHostname represents a custom hostname in a zone.
 type CustomHostname struct {
-	ID             string            `json:"id,omitempty"`
-	Hostname       string            `json:"hostname,omitempty"`
-	SSL            CustomHostnameSSL `json:"ssl,omitempty"`
-	CustomMetadata CustomMetadata    `json:"custom_metadata,omitempty"`
+	ID       string `json:"id,omitempty"`
+	Hostname string `json:"hostname,omitempty"`
+	// CustomOriginServer allows you to override the backend server on a per-hostname basis. This is a optional, added feature.
```
Comment (jamesog) [inline custom_hostname.go]: Nit: "This is an optional" (not "a optional")

**Comment by ElvinEfendi** [comment]
@elithrar when do you expect to complete and merge this?

**Comment by elithrar** [comment]
I no longer work at Cloudflare, and thus won't be completing it (as I don't
use Cloudflare for my day job either).

**Pull request closed without merging**